v2x/vlog_to_pbtype.py

```diff
@@ -54,6 +54,27 @@
     - `(* carry = ADDER *)` : specify carry chain pack_pattern associated
                               with this port
 
+The following attributes are used to annotate cells with fasm metadata:
+    - `(* FASM_PREFIX="prefix" *)` : Sets the fasm prefix of a cell instance.
+        Cannot be set on a module definition!
+
+    - `(* FASM_PREFIX="prefix1;prefix2;..." *) : Same as FASM_PREFIX but used
+        for cells inside `generate` statements. Each instance of a cell will be
+        assigned one prefix from the semicolon separated list. The number of
+        prefixes in the list must match the cell instance count!
+
+    - `(* FASM_FEATURES="str1;str2;..." *)` : Specifies a list of fasm features
+        for the given module. Has to be provided at a module definition.
+        Multiple features have to be separated by semicolon.
+
+    - `*( FASM_FEATURES_{mode}="str1;str2;..." *)` : Same as above but defines
```

Comment:
I think we need to rework how this one works.

Reply:
So the use case here is that we need to emit a feature only in specific mode that is not directly related to any of child cells.

For example: we have a module that holds either 1xSRL32 or 2xSRL16 - these are different modes. When using 2xSRL16 we need to emit the `SMALL` feature. We could specify it in the feature list for both child SRL16s but instead we can bind it to the mode itself.

Reply:
Example please?

Reply:
@mkurc-ant - ping?

Reply:
Another example from XC7: There is the ILOGICE3 site which can host ISERDES, IDDR or nothing for a passthrough mode. In the passthrough mode there has to be a fasm feature emitted which configures one of muxes.

So the site is modeled as a pb_type with 3 modes: PASSTHROUGH, ISEDRES_NO_IDELAY, ISERDES_IDELAY and IDDR (will be). In the PASSTHROUGH mode there are no leaf pb_types, just an interconnect. The required feature is attached to one of its connections. See https://github.com/SymbiFlow/symbiflow-arch-defs/blob/9d0b8f25beec7073345a232b7e8512be6d5e8652/xc/common/primitives/ilogice3/ilogice3.pb_type.xml#L33

Ideally the fasm feature should be attached to an assign statement modeling the interconnect. But unfortunately Yosys does not support that.

The problem is not relevant eg. in the QuickLogic architecture but it will hit us eventually.

Reply:
@mithro Thoughts?

Reply:
I think we should only add the feature when an example like that is added as a test.